src/librustc/traits/select.rs

```diff
@@ -609,11 +630,12 @@ impl<'cx, 'gcx, 'tcx> SelectionContext<'cx, 'gcx, 'tcx> {
                 // does this code ever run?
                 match self.infcx.subtype_predicate(&obligation.cause, obligation.param_env, p) {
                     Some(Ok(InferOk { obligations, .. })) => {
-                        self.evaluate_predicates_recursively(previous_stack, &obligations);
-                        EvaluatedToOk
+                        // FIXME: why are we ignoring the result here?
+                        let _ = self.evaluate_predicates_recursively(previous_stack, &obligations);
```

Comment:
I think that's probably a bug, actually; we used to not even invoke `evaluate_predicates_recursively`, but you added the call in:

81f0b962f37952179e4402a6078ea48f27fc77ad

I think we ought to be returning that result

Reply:
Oops, makes sense, will fix.